misc/python/materialize/cloudtest/k8s/__init__.py

```diff
@@ -37,11 +37,16 @@
 
 
 class K8sResource:
-    def kubectl(self, *args: str, input: Optional[str] = None) -> str:
+    def kubectl(
+        self, *args: str, input: Optional[str] = None, namespace: Optional[str] = None
+    ) -> str:
         try:
-            return subprocess.check_output(
-                ["kubectl", "--context", self.context(), *args], text=True, input=input
-            )
+            cmd = ["kubectl", "--context", self.context(), *args]
```

Comment:
so context is taken from `self`, but the namespace is taken from the argument? Can those two be made orthogonal, that is, methods of `self`?


I must admit I did not design it correctly with two separate classes having a `kubectl` method. Feel free to refactor this if it bothers you.

Reply:
I was indeed slightly confused by the API which allows specifying the pod name so that you could specify an unrelated pod. Anyway, yes, it makes sense to use the stored context here. Will adjust.

Reply:
Addressed in https://github.com/MaterializeInc/materialize/pull/21064.